ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/deploy_cv_pathfinder_metadata_to_cv.py

```diff
@@ -23,6 +25,47 @@
 }
 
 
+class StudioVersion(Enum):
+    """Supported cv-pathfinder metadata studio versions"""
+
+    v3_2 = 3.2
+    v3_3 = 3.3
+    v3_4 = 3.4
+
+
+FEATURE_MIN_VERSION = {
+    "pathfinder_location": StudioVersion.v3_4,
+    "internet_exit_zscaler": StudioVersion.v3_4,
+}
+FEATURE = Literal["pathfinder_location", "internet_exit_zscaler"]
+
+
+async def get_metadata_studio_version(result: DeployToCvResult, cv_client: CVClient) -> StudioVersion | None:
+    """
+    Extract the version from the metadata studio.
+    This is stored as a default value on the studioVersion field.
+    """
+    studio = await cv_client.get_studio(CV_PATHFINDER_METADATA_STUDIO_ID, result.workspace.id)
+    studio_version_field = get(studio.input_schema.fields.values, "studioVersion")
+    version = studio_version_field.string_props.default_value
+    LOGGER.info("deploy_cv_pathfinder_metadata_to_cv: Found metadata studio version: %s", version)
+    try:
+        return StudioVersion(float(version))
+    except ValueError:
+        accepted_versions = [str(v.value) for v in StudioVersion]
+        warning = (
+            f"deploy_cv_pathfinder_metadata_to_cv: Got invalid metadata studio version '{version}'. "
+            f"This plugin only accepts versions '{accepted_versions}'. Skipping upload of metadata."
+        )
```

Comment:
So this means that we are locking each AVD version to a limited set of studio versions?

Reply:
Yes. We have no way of filling out the correct fields otherwise.

Reply:
Changed to check supported fields in the schema instead. Thanks.

Reply:
don't we prefer to think that we are non breaking going forward and just adding things in this metadata studio? This is the same discussion we had regarding having dependencies locked at some version - it will bites us in the future

Reply:
ok my comment came too late :)